### PR TITLE
Add PGCMP Workflow

### LINKED ISSUE #567: Use pgcmp to compare migrations
In the interest of doing a more nuanced diff than just "pg_dump -> string comparison", we are looking into using [pgcmp](https://github.com/cbbrowne/pgcmp).

Current use case: Make sure that the schema that results from running all the migrations matches the latest schema.

Future use case: Provide the users with a way to detect database drift.

Source discussion: #427 

Two possible approaches are:
1. As of #427, the "pg_dump -> string comparison" test is in the `db-tests` suite - the pgcmp test could replace it
2. Alternatively, the pgcmp test could be split into a separate github workflow

### PR DESCRIPTION
**Pull request by Mythicaeda**
* **Tickets addressed:** Closes #567
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description

Added a workflow to:
1) Build the version of the DB from 1.0.1 (the base version for migrations), migrate it to current, then dump it
2) Build the version of the DB on the current branch, then dump it
3) Use `pgcmp` to compare the dumps for equality
4) Upload the results of `pgcmp` should at least one comparison fail.

## Verification

Successful running on this branch should indicate if it's working.

## Future work

Potentially refine the script to only look at the changes between develop and the migration defined in the PR.

### CODE REVIEW

File: .github/workflows/pgcmp.yml [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+on:
+  pull_request:
+    paths:
+      - 'deployment/hasura/migrations/**'
+      - 'merlin-server/sql/**'
+      - 'sequencing-server/sql/**'
+      - 'scheduler-server/sql/**'
+      - 'deployment/postgres-init-db/sql/**'
+      - '.github/**' # REMOVE BEFORE MERGING, BUT THIS SHOULD FORCE THE WORKFLOW TO RUN ON THIS PR
```
Comment (cohansen) [inline .github/workflows/pgcmp.yml]: Just tagging this so it's not forgotten :)

**Review by cohansen — CHANGES_REQUESTED** [review]
Looks fine to me, I ran locally and Aerie ran successfully.

It looks like there's some failed checks though, and 1 remove before merge.

**Comment by Mythicaeda** [comment]
> Looks fine to me, I ran locally and Aerie ran successfully.
You ran the workflow locally or you deployed Aerie from this branch successfully? Double-checking because this PR is adding a workflow to Aerie, not modifying any existing Aerie code

**Comment by cohansen** [comment]
> > Looks fine to me, I ran locally and Aerie ran successfully.
> > You ran the workflow locally or you deployed Aerie from this branch successfully? Double-checking because this PR is adding a workflow to Aerie, not modifying any existing Aerie code

Ah, I misunderstood your instructions. I ran Aerie locally which I guess doesn't test this workflow. My comment above was because I saw some of the checks had failed against this branch.

**Comment by Mythicaeda** [comment]
> Ah, I misunderstood your instructions. I ran Aerie locally which I guess doesn't test this workflow. My comment above was because I saw some of the checks had failed against this branch.

Yeah, that failing check is actually the one being added

File: .github/workflows/pgcmp.yml (hunk 1 above)
Comment (Mythicaeda) [reply .github/workflows/pgcmp.yml]: It's been removed!

**Review by cohansen — APPROVED** [review]
Looks great and the checks passed! 🎉

**Pull request merged**